src/test/ui/__check/issue-63695-1.rs

```diff
@@ -0,0 +1,19 @@
+// run-pass
+#![feature(const_generics)]
+//~^ WARN the feature `const_generics` is incomplete and may cause the compiler to crash
+
+fn test<const A: i32>() -> i32 { A }
+
+trait T {
+    fn test<const A: i32>(&self) -> i32 { A }
+}
+
+struct S();
+
+impl T for S {}
+
+fn main() {
+  let foo = S();
+  assert_eq!(test::<{8i32}>(), 8);
+  assert_eq!(foo.test::<{16i32}>(), 16);  // <- Causes ICE
```

Comment:
This is a method call example, this is the kind of thing that's impossible to support for now, as I've mentioned in at least one issue.